src/mainwindow.cpp

```diff
@@ -5776,6 +5777,20 @@ void MainWindow::nxmRequestFailed(QString gameName, int modID, int, QVariant, in
 {
   if (error == QNetworkReply::ContentAccessDenied || error == QNetworkReply::ContentNotFoundError) {
     log::debug("{}", tr("Mod ID %1 no longer seems to be available on Nexus.").arg(modID));
+    
+    // update last checked timestamp on orphaned mods as well to avoid repeating requests
+    QString gameNameReal;
+    for (IPluginGame* game : m_PluginContainer.plugins<IPluginGame>()) {
+      if (game->gameNexusName() == gameName) {
+        gameNameReal = game->gameShortName();
+        break;
+      }
+    }
```

Comment:
I'm not sure this should be here... Or maybe default `gameNameReal` to `gameName`? It's kind of a mess trying to understand the origin of `gameName`, but in some case it seems to be `m_GameName` from the `ModInfo` directly (e.g. in `ModInfoRegular::updateNXMInfo`) which should be used for `getByModId`.

Reply:
Ok, forget this comment... You pass `m_GameName` in `requestDescription` but you actually fetch the corresponding plugin from it and use `gameNexusName()` 🙃 This naming stuff really needs some refactoring to make it clearer 🙄

Reply:
yeah, it got to this point because nexus changed the names they were using so instead of gameShortName we had to add gameNexusName when before we were doing everything with gameShortName. We couldn't change the gameShortName of games since we use it for persistence of all kinds of things and it would have broken setups.

Reply:
Yes, I think that one issue is that `ModInfo` uses `GameName` for the nexus game name in practice... Maybe one thing would be to wrap the "origin" of a mod in a structure (e.g., for `ModInfo` / Nexus it would be mod ID + nexus game name) so that you could pass that structure around and it would be clear what those means. That's something we'll probably have to do one if we want to move the nexus stuff to a separate plugin.

Reply:
I don't really feel any enthusiasm for that prospective